### PR TITLE
feat(gtrack): inplace output

### PR DESCRIPTION
**Pull request by s22chan**
useful for reducing copying overhead when merging tracks

### CODE REVIEW

**Comment by s22chan** [comment]
@ovesh is on vacation so please review the usage and approve if OK

File: genome_kit/genome_track.py [hunk 1]
```diff
@@ -568,6 +568,8 @@ def __call__(self, interval, dtype=None):  # pragma: no cover
             The stranded query interval.
         dtype : :py:class:`type`
             Optional. The numpy dtype of the resulting array.
+        dtype : :py:class:`~numpy.ndarray`
```
Comment (alstonlo) [inline genome_kit/genome_track.py:571]: I think there is a minor typo here! 
```suggestion
        out : :py:class:`~numpy.ndarray`
```

**Review by alstonlo — APPROVED** [review]
Just one question out of curiosity: how is the case handled where the `dtype` of `out` is incompatible with that of the GenomeTrack?

**Comment by s22chan** [comment]
> Just one question out of curiosity: how is the case handled where the `dtype` of `out` is incompatible with that of the GenomeTrack?

It should throw since no decoder was registered:
https://github.com/deepgenomics/GenomeKit/blob/1e2b67222fd3504dfd112925893f633bf665f11c/src/genome_track.h#L1073-L1086

**Pull request merged**